File: Semester5/Mobile Applications/flutter/books_tbr/lib/model/Book.dart
import 'package:scoped_model/scoped_model.dart';

enum Status {
  UNREAD('UNREAD'), IN_PROGRESS('IN_PROGRESS'), READ('READ');
  const Status(this.value);
  final String value;
}

class Book extends Model {
  int id;
  String title;
  String author;
  int nrPages;
  Status status;
  String genre;

  Book(this.id, this.title, this.author, this.nrPages, this.status, this.genre);
}

File: Semester5/Mobile Applications/android/app/src/main/java/oana/books/bookstbr/books/model/Book.kt
package oana.books.bookstbr.books.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "book_table")
data class Book(
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") var id: Int,
    @ColumnInfo(name = "title") val title: String,
    @ColumnInfo(name = "author") val author: String,
    @ColumnInfo(name = "nrPages") val nrPages: Int,
    @ColumnInfo(name = "status") val status: Status,
    @ColumnInfo(name = "genre") val genre: String
)
